extensions/database-documentation/src/common/utils.ts

```diff
@@ -0,0 +1,179 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata'
+import * as vscode from 'vscode';
+// Language localization features
+import * as nls from 'vscode-nls';
+// The module 'openai' contains the OpenAI API
+import { Configuration, OpenAIApi } from "openai";
+
+const localize = nls.loadMessageBundle();
+
+export async function generateMarkdown(context: azdata.ObjectExplorerContext): Promise<string> {
+	let connection = (await azdata.connection.getCurrentConnection());
+	let databaseName = context.connectionProfile.databaseName;
+
+	let tables: string[];
+	if (context.nodeInfo.nodeType === 'Table') {
+		tables = [context.nodeInfo.metadata.name];
+	}
+	else {
+		const queryProvider = azdata.dataprotocol.getProvider<azdata.QueryProvider>("MSSQL", azdata.DataProviderType.QueryProvider);
+		const connectionUri = await azdata.connection.getUriForConnection(connection.connectionId);
+		let query = `SELECT TABLE_NAME FROM [${databaseName}].INFORMATION_SCHEMA.TABLES WHERE TABLE_TYPE = 'BASE TABLE'`;
+
+		if (context.nodeInfo.nodeType === 'Schema') {
+			query += ` AND TABLE_SCHEMA = '${context.nodeInfo.metadata.name}'`;
+		}
+
+		tables = (await queryProvider.runQueryAndReturn(connectionUri, query)).rows.map(row => row[0].displayValue);
+	}
+
+	let diagram = '```mermaid\nclassDiagram\n';
+	let references = ``;
+	let documentation = ``;
+	for (let i = 0; i < tables.length; i++) {
+		const tableAttributes = await tableToText(connection, databaseName, tables[i]);
+		const tableResult = getMermaidDiagramForTable(tables[i], tableAttributes);
+		diagram += tableResult[0] + `\n`;
+		if (context.nodeInfo.nodeType !== 'Table') {
+			references += tableResult[1];
+		}
+		documentation += await getDocumentationText(tables[i], tableAttributes);
+	}
+
+	diagram += references;
+
+	return diagram + '```  \n\n' + documentation;
+}
+
+export async function tableToText(connection: azdata.connection.ConnectionProfile, databaseName: string, tableName: string): Promise<[string, string, string][]> {
+
+	let queryProvider = azdata.dataprotocol.getProvider<azdata.QueryProvider>("MSSQL", azdata.DataProviderType.QueryProvider);
+
+	let connectionUri = await azdata.connection.getUriForConnection(connection.connectionId);
+	let baseQuery = `SELECT COLUMN_NAME, DATA_TYPE FROM [${databaseName}].INFORMATION_SCHEMA.COLUMNS WHERE TABLE_NAME = '${tableName}'`;
```

Comment:
do you know if something like `table;DROP table1` is a valid table name?

Reply:
I already have https://github.com/microsoft/azuredatastudio/pull/23409#discussion_r1235923056 for fixing potential SQL injection risks